### PR TITLE
Fix IndexError in para_split_v3.py for empty line handling

### PR DESCRIPTION
**Pull request by hyastar**
This PR resolves an IndexError in the __is_list_or_index_block function that occurs when lines_text_list contains an empty string. Changes include:

- Added an empty string check before accessing the last character.
- Replaced direct indexing with safer slicing (text[-1:] instead of text[-1]).
- Enhanced handling for empty or whitespace-only lines.

Sample input that caused the error: ['a.', 'b.', 'c.', 'd.', 'e.', 'f.', '']

### CODE REVIEW

**Comment by hyastar** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by hyastar** [comment]
recheck

**Comment by hyastar** [comment]
### Additional Information

When processing large PDF files (280MB or larger) using the `magic-pdf` library, an `IndexError` frequently occurs in the `__is_list_or_index_block` function within `para_split_v3.py`. This error is encountered in an environment with an RTX 4090 GPU, 100GB of system memory, and 24GB of GPU memory when the `lines_text_list` contains an empty string, causing an out-of-range error.

### Test Environment

- **GPU**: NVIDIA RTX 4090
- **System Memory**: 100GB
- **GPU Memory**: 24GB

File: magic_pdf/para/para_split_v3.py [hunk 1]
```diff
@@ -102,7 +102,9 @@ def __is_list_or_index_block(block):
                 if span_type == ContentType.Text:
                     line_text += span['content'].strip()
 
-            lines_text_list.append(line_text)
+            # 只添加非空文本
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py]: 空行也要添加进去，因为lines_text_list的长度要和block['lines']保持一致，如果长度不一致，后面用index匹配就会出现错位

File: magic_pdf/para/para_split_v3.py [hunk 2]
```diff
@@ -128,21 +130,23 @@ def __is_list_or_index_block(block):
         num_start_count = 0
         num_end_count = 0
         flag_end_count = 0
+
         if len(lines_text_list) > 0:
             for line_text in lines_text_list:
-                if len(line_text) > 0:
-                    if line_text[-1] in LIST_END_FLAG:
+                if len(line_text) > 0:  # 额外检查确保不是空字符串
+                    if line_text and line_text[-1] in LIST_END_FLAG:
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py]: line_text>0时，不需要再通过if line_text判断是否为空

File: magic_pdf/para/para_split_v3.py [hunk 3]
```diff
@@ -128,21 +130,23 @@ def __is_list_or_index_block(block):
...
-                    if line_text[-1].isdigit():
+                    if line_text and line_text[-1].isdigit():
                         num_end_count += 1
 
-            if flag_end_count / len(lines_text_list) >= 0.8:
-                line_end_flag = True
-
-            if num_start_count / len(lines_text_list) >= 0.8 or num_end_count / len(lines_text_list) >= 0.8:
-                line_num_flag = True
+            total_valid_lines = len(lines_text_list)
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py]: 这边处在`if len(lines_text_list) > 0:`的内部，lines_text_list的长度一定是大于0的，所以不需要判断长度是否为0

File: magic_pdf/para/para_split_v3.py [hunk 4]
```diff
@@ -176,7 +180,7 @@ def __is_list_or_index_block(block):
                 # 这种是大部分line item 都有结束标识符的情况，按结束标识符区分不同item
                 elif line_end_flag:
                     for i, line in enumerate(block['lines']):
-                        if lines_text_list[i][-1] in LIST_END_FLAG:
+                        if i < len(lines_text_list) and lines_text_list[i] and lines_text_list[i][-1] in LIST_END_FLAG:
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py]: 这里原方法少了对lines_text_list[i]是否为空做判断，可能会有error，可以直接改成
```python
if len(lines_text_list[i]>0) and lines_text_list[i][-1] in LIST_END_FLAG:
```

因为lines_text_list的长度和block['lines']一致，所以不需要判断`i<len(lines_text_list)`

File: magic_pdf/para/para_split_v3.py [hunk 5]
```diff
@@ -188,16 +192,19 @@ def __is_list_or_index_block(block):
...
                             line[ListLineTag.IS_LIST_END_LINE] = True
                             line_start_flag = True
             # 一种有缩进的特殊有序list,start line 左侧不贴边且以数字开头，end line 以 IS_LIST_END_LINE 结尾且数量和start line 一致
             elif num_start_count >= 2 and num_start_count == flag_end_count:  # 简单一点先不考虑左侧不贴边的情况
                 for i, line in enumerate(block['lines']):
-                    if lines_text_list[i][0].isdigit():
-                        line[ListLineTag.IS_LIST_START_LINE] = True
-                    if lines_text_list[i][-1] in LIST_END_FLAG:
-                        line[ListLineTag.IS_LIST_END_LINE] = True
+                    if i < len(lines_text_list) and lines_text_list[i]:
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py]: 同理，不需要这个判断

File: magic_pdf/para/para_split_v3.py [hunk 6]
```diff
@@ -188,16 +192,19 @@ def __is_list_or_index_block(block):
...
+                        
+                        if lines_text_list[i][0].isdigit():
```
Comment (myhloli) [inline magic_pdf/para/para_split_v3.py:195]: 这里类似上面, 少了内容是否为空的判断，加一个len的检测就好
```python
if len(lines_text_list[i]>0) and lines_text_list[i][0].isdigit()::
```

**Review by myhloli — COMMENTED** [review]
另外建议通过拉取dev分支的方式提交自己的代码，并提交pr到dev分支，因为我们的开发都会在dev分支完成，并仅在release时同步到master

**Pull request closed without merging**